### ISSUE TITLE
Quick fix context menu only displays “consider” for `unreadable-literal` warnings

### ISSUE THREAD

**Issue by EFanZh** [rust-analyzer/rust-analyzer#7169]
See the following screenshot:

![image](https://user-images.githubusercontent.com/2160692/103645681-5f255100-4f93-11eb-97ad-9148cc877d3c.png)

Can it be something like “consider: `1_000_000_000`”?

OS: Windows 10.

**Comment by flodiebold** [comment]
That's probably a bug with the labeling in rustc/clippy.

**Comment by jonas-schievink** [comment]
@EFanZh What does `cargo clippy` output on the terminal for this?

**Comment by EFanZh** [comment]
Clippy seems file:

```
warning: long literal lacking separators
 --> src\main.rs:2:13
  |
2 |     let _ = 1000000000;
  |             ^^^^^^^^^^ help: consider: `1_000_000_000`
  |
  = note: `-W clippy::unreadable-literal` implied by `-W clippy::pedantic`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unreadable_literal

warning: 1 warning emitted

    Finished dev [unoptimized + debuginfo] target(s) in 0.03s
```

**Comment by lnicola** [comment]
We're getting this:

```json
{
    "reason": "compiler-message",
    "package_id": "hello 0.1.0 (path+file:///C:/Users/Lauren%C8%9Biu/Source/hello)",
    "target": {
        "kind": [
            "bin"
        ],
        "crate_types": [
            "bin"
        ],
        "name": "hello",
        "src_path": "C:\\Users\\Laurențiu\\Source\\hello\\src\\main.rs",
        "edition": "2018",
        "doctest": false,
        "test": true
    },
    "message": {
        "rendered": "error: long literal lacking separators\n --> src\\main.rs:5:16\n  |\n5 |     let _foo = 1000000000;\n  |                ^^^^^^^^^^ help: consider: `1_000_000_000`\n  |\nnote: the lint level is defined here\n --> src\\main.rs:1:9\n  |\n1 | #![deny(clippy::clippy::clippy::unreadable_literal)]\n  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unreadable_literal\n\n",
        "children": [
            {
                "children": [],
                "code": null,
                "level": "note",
                "message": "the lint level is defined here",
                "rendered": null,
                "spans": [
                    {
                        "byte_end": 50,
                        "byte_start": 8,
                        "column_end": 51,
                        "column_start": 9,
                        "expansion": null,
                        "file_name": "src\\main.rs",
                        "is_primary": true,
                        "label": null,
                        "line_end": 1,
                        "line_start": 1,
                        "suggested_replacement": null,
                        "suggestion_applicability": null,
                        "text": [
                            {
                                "highlight_end": 51,
                                "highlight_start": 9,
                                "text": "#![deny(clippy::clippy::clippy::unreadable_literal)]"
                            }
                        ]
                    }
                ]
            },
            {
                "children": [],
                "code": null,
                "level": "help",
                "message": "for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unreadable_literal",
                "rendered": null,
                "spans": []
            },
            {
                "children": [],
                "code": null,
                "level": "help",
                "message": "consider",
                "rendered": null,
                "spans": [
                    {
                        "byte_end": 122,
                        "byte_start": 112,
                        "column_end": 26,
                        "column_start": 16,
                        "expansion": null,
                        "file_name": "src\\main.rs",
                        "is_primary": true,
                        "label": null,
                        "line_end": 5,
                        "line_start": 5,
                        "suggested_replacement": "1_000_000_000",
                        "suggestion_applicability": "MachineApplicable",
                        "text": [
                            {
                                "highlight_end": 26,
                                "highlight_start": 16,
                                "text": "    let _foo = 1000000000;"
                            }
                        ]
                    }
                ]
            }
        ],
        "code": {
            "code": "clippy::unreadable_literal",
            "explanation": null
        },
        "level": "error",
        "message": "long literal lacking separators",
        "spans": [
            {
                "byte_end": 122,
                "byte_start": 112,
                "column_end": 26,
                "column_start": 16,
                "expansion": null,
                "file_name": "src\\main.rs",
                "is_primary": true,
                "label": null,
                "line_end": 5,
                "line_start": 5,
                "suggested_replacement": null,
                "suggestion_applicability": null,
                "text": [
                    {
                        "highlight_end": 26,
                        "highlight_start": 16,
                        "text": "    let _foo = 1000000000;"
                    }
                ]
            }
        ]
    }
}
```

Notice how the inner message actually says `consider`, but we don't display the outer one.

Duplicate of #6851, I guess? CC @jonas-schievink.

**Comment by bjorn3** [comment]
> Notice how the inner message actually says consider, but we don't display the outer one.

There could be multiple fixes for the same error, which would each get their own sub-diagnostic within the error diagnostic, so we shouldn't use the message of the outer diagnostic.

**Comment by lnicola** [comment]
Hmm, why are we showing `consider` twice, if it's only once in the JSON?

**Comment by lnicola** [comment]
Anyway, I think we can close this. We're showing the message as reported by `clippy`.

**Comment by EFanZh** [comment]
So is this a bug from clippy? Should I open a bug report there?

**Comment by lnicola** [comment]
It seems like an unfortunate design: the `clippy` message is only `consider`, but when printing the warning as text it includes the fixed code. We don't do that because the showing the replacements isn't really needed for `rustc` warnings (and I'm not sure if they're always single-line or not).
